### PR TITLE
Crypto Performance Improvements

### PR DESCRIPTION
**Pull request by lgrahl**
This PR intends to improve the overall encryption/decryption performance. It also adds a performance test suite.

Results from the new tests so far are very promising. The base (*Main Thread (shared key store=false)*) is what we used before this PR.

Results are from an AMD Ryzen 7 2700X (higher % is faster).

| **Encryption**                                           | Chrome 67.0.3396.99 | Firefox 61.0.1 |
|----------------------------------------------------------|---------------------|----------------|
| **Main Thread (shared key store=false)**                     | 100% (base)         | 100% (base)    |
| Main Thread (shared key store=true)                      | 201%                | 281%           |
| Web Worker (shared key store=false, transferables=false) | 93%                 | 95%            |
| Web Worker (shared key store=false, transferables=true)  | 96%                 | 97%            |
| Web Worker (shared key store=true, transferables=false)  | 171%                | 239%           |
| Web Worker (shared key store=true, transferables=true)   | 176%                | 245%           |

| **Decryption**                                           | Chrome 67.0.3396.99 | Firefox 61.0.1 |
|----------------------------------------------------------|---------------------|----------------|
| **Main Thread (shared key store=false)**                 | 100% (base)         | 100% (base)    |
| Main Thread (shared key store=true)                      | 197%                | 275%           |
| Web Worker (shared key store=false, transferables=false) | 91%                 | 93%            |
| Web Worker (shared key store=false, transferables=true)  | 92%                 | 96%            |
| Web Worker (shared key store=true, transferables=false)  | 161%                | 235%           |
| Web Worker (shared key store=true, transferables=true)   | 174%                | 254%           |

To Do:
- [x] Use `SharedKeyStore` by default in the client
- [x] Rebase against master
- [x] Add tests for `SharedKeyStore`
- [x] ~~Update docs (need a local server for the performance tests)~~
- [x] Evaluate whether depending task implementations need to be updated to leverage `SharedKeyStore` (answer: no)

### CODE REVIEW

File: tests/keystore.spec.ts [hunk 1]
```diff
@@ -156,6 +156,80 @@ export default () => { describe('keystore', function() {
...
+
+            encrypted = sks.encrypt(expected, nonce);
+            expect(sks.decrypt(encrypted)).toEqual(expected);
+            expect(sks.decryptRaw(encrypted.data, encrypted.nonce)).toEqual(expected);
+
+            encrypted = sks.encryptRaw(expected, nonce);
+            const encryptedBox = new Box(nonce, encrypted, nacl.box.nonceLength);
+            expect(sks.decrypt(encryptedBox)).toEqual(expected);
+            expect(sks.decryptRaw(encrypted, nonce)).toEqual(expected);
+        });
```
Comment (dbrgn) [inline tests/keystore.spec.ts]: You could also add a test where you encrypt/decrypt using the shared keystore and also with the regular keystore, and ensure that the results are the same :)

**Comment by dbrgn** [comment]
Also, before merging this, could you squash the two cleanup commits "Fix lint errors" and "Cleanup let -> const in keystore.spec.ts" into their preceding commits? They don't add any value as standalone commits :slightly_smiling_face:

File: tests/keystore.spec.ts (hunk 1 above)
Comment (lgrahl) [reply tests/keystore.spec.ts]: Done!

**Comment by dbrgn** [comment]
I fixed the [remaining issue](https://github.com/saltyrtc/saltyrtc-client-js/pull/99#discussion_r204795633). Will merge once tests pass.

**Pull request merged**